src/main/java/com/kneelawk/extramodintegrations/tconstruct/recipe/melting/MeltingEmiRecipe.java

```diff
@@ -32,18 +38,23 @@ public static MeltingEmiRecipe of(MeltingRecipe recipe) {
         } else {
             id = recipe.getId();
         }
-        return new MeltingEmiRecipe(recipe, id);
+        return new MeltingEmiRecipe(recipe, id, Config.COMMON.smelteryOreRate);
     }
 
-    private MeltingEmiRecipe(MeltingRecipe recipe, Identifier id) {
+    private MeltingEmiRecipe(MeltingRecipe recipe, Identifier id, IMeltingContainer.IOreRate oreRate) {
         super(TiCCategories.MELTING, id);
 
-        this.inputs = recipe.getIngredients().stream().map(EmiIngredient::of).toList();
-        this.outputs = List.of(Util.convertFluid(recipe.getOutput()));
-
         this.time = recipe.getTime();
         this.temperature = recipe.getTemperature();
         this.oreRateType = recipe.getOreType();
+
+        this.inputs = recipe.getIngredients().stream().map(EmiIngredient::of).toList();
+        FluidStack originalOutput = recipe.getOutput();
+        if (this.oreRateType != null) {
+            originalOutput = oreRate.applyOreBoost(oreRateType, originalOutput);
+        }
+        this.outputs = List.of(Util.convertFluid(originalOutput));
+        this.outputsTiCTooltip = List.of(Util.getFluidTiCTooltip(originalOutput));
```

Comment:
Is this stuff applicable to the more general `AbstractMeltingEmiRecipe`? Could this be generalized more?

Reply:
Right now its only being applicable  for `MeltingEmiRecipe`, because FoundaryEMIRecipe  calls `getOutputWithByproducts` to get output, and within it tcon calculates rates. So its not required there. It's not applicable for Entity Melting. Doesn't apply to Alloy too (which doesn't even inherit from `AbstractMeltingEmiRecipe`).